src/microbe_stage/Microbe.Contact.cs

```diff
@@ -430,18 +579,153 @@ public void SendOrganellePositionsToMembrane()
+
+        return OnKilled();
+    }
+
+    public void OnDestroyed()
+    {
+        if (destroyed)
+            return;
+
+        destroyed = true;
+
+        // TODO: find out a way to cleanly despawn colonies without having to run the reproduction progress lost logic
+        Colony?.RemoveFromColony(this);
+
+        AliveMarker.Alive = false;
+    }
+
+    /// <summary>
+    ///   Removes this cell and child cells from the colony.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     If this is the colony master, this disbands the whole colony
+    ///   </para>
+    /// </remarks>
+    public void UnbindAll()
+    {
+        if (State is MicrobeState.Unbinding or MicrobeState.Binding)
+            State = MicrobeState.Normal;
+
+        if (!CanUnbind)
+            return;
+
+        // TODO: once the colony leader can leave without the entire colony disbanding this perhaps should keep the
+        // disband entire colony functionality
+        Colony!.RemoveFromColony(this);
+    }
+
+    internal void OnColonyMemberRemoved(Microbe microbe)
+    {
+        cachedColonyRotationMultiplier = null;
+
+        if (microbe == this)
+        {
+            OnUnbound?.Invoke(this);
+
+            RevertNodeParent();
+            ai?.ResetAI();
+
+            Mode = ModeEnum.Rigid;
+
+            return;
+        }
+
+        if (IsMulticellular && Colony?.Master == this)
+        {
+            // Lost a member of the multicellular organism
+            OnMulticellularColonyCellLost(microbe);
+        }
+
+        if (HostileEngulfer != microbe)
+            microbe.RemoveCollisionExceptionWith(this);
+        if (microbe.HostileEngulfer != this)
+            RemoveCollisionExceptionWith(microbe);
+    }
+
+    internal void ReParentShapes(Microbe to, Vector3 offset)
+    {
+        // TODO: if microbeRotation is the rotation of *this* instance we should use the variable here directly
+        // An object doesn't need to be told its own member variable in a method...
+        // https://github.com/Revolutionary-Games/Thrive/issues/2504
+        foreach (var organelle in organelles!)
+            organelle.ReParentShapes(to, offset);
+    }
+
+    internal void OnColonyMemberAdded(Microbe microbe)
+    {
+        cachedColonyRotationMultiplier = null;
+
+        if (microbe == this)
+        {
+            if (Colony == null)
+            {
+                throw new InvalidOperationException(
+                    $"{nameof(Colony)} must be set before calling {nameof(OnColonyMemberAdded)}");
+            }
+
+            OnIGotAddedToColony();
+
+            if (Colony.Master != this)
+            {
+                Mode = ModeEnum.Static;
+            }
+
+            ReParentShapes(Colony.Master, GetOffsetRelativeToMaster());
+        }
+        else
+        {
+            AddCollisionExceptionWith(microbe);
+            microbe.AddCollisionExceptionWith(this);
+        }
+    }
+
+    internal void SuccessfulScavenge()
```

Comment:
Seems like a bunch of methods have been moved around, making the diff hard to read, is there a reason for this?
I'll try to read only the actually changed parts later in this file...

Reply:
Because I separated half of what's in `Kill` to another method, it is private and so is placed below `internal void` methods hence the cause for these diff very likely.

Reply:
That explains it, though that's basically impossible to see in the Github diff, I'll need to use my IDE to separately read through the microbe contact file to see what it looks like after the changes.